core/shared/src/main/scala/zio/internal/metrics/ConcurrentMetricRegistry.scala

```diff
@@ -18,32 +18,55 @@ private[zio] class ConcurrentMetricRegistry {
 
   def snapshot()(implicit unsafe: Unsafe): Set[MetricPair.Untyped] = {
     val iterator = map.entrySet().iterator()
-    val result   = scala.collection.mutable.Set[MetricPair.Untyped]()
+    val result   = Set.newBuilder[MetricPair.Untyped]
+    result.sizeHint(map.size())
```

Comment:
@plokhotnyuk Is the `ConcurrentHashMap::size` call too expensive to justify adding this line? 🤔